oak-store-spi/src/main/java/org/apache/jackrabbit/oak/plugins/memory/AbstractBlob.java

```diff
@@ -189,4 +182,39 @@ public String toString() {
         return getSha256().toString();
     }
 
+    public static class HashCode {
+
+        private byte[] digest;
+
+        private HashCode(byte[] digest){
+            this.digest = digest;
+        }
+
+        private static HashCode of(byte[] bytes) {
+            try {
+                return new HashCode(MessageDigest.getInstance("SHA-256").digest(bytes));
+            } catch (NoSuchAlgorithmException ex) {
+                throw new IllegalStateException("no SHA256 digest", ex);
+            }
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (o == null || getClass() != o.getClass()) return false;
```

Comment:
That is no good. The actual param could be of a derived class. I'd use "if (!(o instanceof AbstractBlob))"

Reply:
It was generated by IntelliJ :-)